### PR TITLE
document how to migrate between collections

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

Move this section of the collections FAQ to docs - https://github.com/ansible-collections/overview/blob/master/README.rst#q-what-should-i-do-to-move-plugins-across-collections-during-migration

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs.ansible.com
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by acozine — APPROVED** [review]
Looks great, thanks @samccann.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 1]
```diff
@@ -470,24 +470,58 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+Create a PR against the old collection repo to remove the modules, module_utils, plugins, and docs_fragments related to this migration:
+
+#. If you are moving an action plugin, include the corresponding module with documentation.
+#. If you are moving a module, move any corresponding action plugin that should stay with it.
+#. Ensure ``meta/`` has updates to ``action_groups.yml`` and ``runtime.yml`` if they did in step #1.
+#. Remove sanity ignore lines from ``tests/sanity/ignore\*.txt``
+#. Remove associated integration tests from ``tests/integrations/targets/`` and unit tests from ``tests/units/plugins/``.
+#. if you are moving from content from ``community.general``, remove entries from ``.github/BOTMETA.yml``.
+#. Carefully review ``meta/runtime.yml`` for any entries, in particular deprecated entries.
+#. Update ``meta/runtime.yml`` to contain redirects for EVERY PLUGIN, pointing to the new collection name.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst:494]:
```suggestion
#. Update ``meta/runtime.yml`` to contain redirects for EVERY PLUGIN, pointing to the new collection name.
#. If possible, do not yet add deprecation warnings to the new ``meta/runtime.yml`` entries, but only for a later major release. So the order should be:
    1. Remove content, add redirects in 3.0.0;
    2. Deprecate redirects in 4.0.0;
    3. Set removal version to 5.0.0 or later.
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 2]
```diff
@@ -470,24 +470,58 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
 
-See the `content_collector README <https://github.com/ansible/content_collector>`_ for full details and usage guidelines.
+Removing the content from the old collection
+----------------------------------------------
+
+Create a PR against the old collection repo to remove the modules, module_utils, plugins, and docs_fragments related to this migration:
+
+#. If you are moving an action plugin, include the corresponding module with documentation.
+#. If you are moving a module, move any corresponding action plugin that should stay with it.
+#. Ensure ``meta/`` has updates to ``action_groups.yml`` and ``runtime.yml`` if they did in step #1.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: How is this related to step 1?

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 3]
```diff
@@ -470,24 +470,58 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+#. Remove sanity ignore lines from ``tests/sanity/ignore\*.txt``
+#. Remove associated integration tests from ``tests/integrations/targets/`` and unit tests from ``tests/units/plugins/``.
+#. if you are moving from content from ``community.general``, remove entries from ``.github/BOTMETA.yml``.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. if you are moving from content from ``community.general`` or ``community.network``, remove entries from ``.github/BOTMETA.yml``.
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 4]
```diff
@@ -470,24 +470,58 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+#. Carefully review ``meta/runtime.yml`` for any entries, in particular deprecated entries.
+#. Update ``meta/runtime.yml`` to contain redirects for EVERY PLUGIN, pointing to the new collection name.
+
+
+Adding the content to the new collection
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst:509]:
```suggestion
.. warning::

	Maintainers for the old collection have to make sure that the PR is merged in a way that it does not break user experience and semantic versioning:
	#. A new version containing the merged PR must not be released before the collection the content has been moved to has been released again, with that content contained in it. Otherwise the redirects cannot work and users relying on that content will experience breakage.
	#. Once 1.0.0 of the collection from which the content has been removed has been released, such PRs can only be merged for a new **major** version (i.e. 2.0.0, 3.0.0, etc.).


Adding the content to the new collection
```

**Review by felixfontein — COMMENTED** [review]
I've added some recommendations I think are useful. @gundalow @abadger what do you think about them?

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 2 above)
Comment (gundalow) [reply docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. Remove any entries about removed plugins from ``meta/action_groups.yml`` and ``meta/runtime.yml``. Ensure they are added into the new repo.
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 5]
```diff
@@ -470,24 +470,69 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+#. Create a PR against the new collection to add the files removed in step 1.
+#. Update the ``ansible/ansible:devel`` branch entries for all files moved.
+
+
+Removing the content from the old collection
+----------------------------------------------
+
+Create a PR against the old collection repo to remove the modules, module_utils, plugins, and docs_fragments related to this migration:
+
+#. If you are moving an action plugin, include the corresponding module with documentation.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: I had to read this a couple of times before I understood it. How about this (updated to match the next items' phrasing)?
```suggestion
#. If you are moving an action plugin, remove the corresponding module that contains the documentation.
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 6]
```diff
@@ -470,24 +470,69 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+#. If you are moving a module, move any corresponding action plugin that should stay with it.
+#. Remove any entries about removed plugins from ``meta/action_groups.yml`` and ``meta/runtime.yml``. Ensure they are added into the new repo.
+#. Remove sanity ignore lines from ``tests/sanity/ignore\*.txt``
+#. Remove associated integration tests from ``tests/integrations/targets/`` and unit tests from ``tests/units/plugins/``.
+#. if you are moving from content from ``community.general`` or ``community.network``, remove entries from ``.github/BOTMETA.yml``.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. If you are moving from content from ``community.general`` or ``community.network``, remove entries from ``.github/BOTMETA.yml``.
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 7]
```diff
@@ -470,24 +470,70 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+#. Update the ``ansible/ansible:devel`` branch entries for all files moved.
+
+
+Removing the content from the old collection
+----------------------------------------------
+
+Create a PR against the old collection repo to remove the modules, module_utils, plugins, and docs_fragments related to this migration:
+
+#. If you are moving an action plugin, include the corresponding module with documentation.
+#. If you are moving a module, move any corresponding action plugin that should stay with it.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. If you are moving a module, remove any corresponding action plugin that should stay with it.
```

**Review by acozine — COMMENTED** [review]
Found a couple of nits. Thanks for this @samccann.

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 5 above)
Comment (samccann) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @gundalow @felixfontein please review this edit. I think it should be 'move' not 'remove'

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 7 above)
Comment (samccann) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @felixfontein @gundalow same question - I think it should stay as move.

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 6 above)
Comment (samccann) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @felixfontein @gundalow - Alicia brought up some good points in discussing this with her.  This last four talk about removing items but aren't necessarily clear if you remove and forget, or if you are actually moving them, or redesigning them in the 2nd pr.  Can you help clarify this section?  for example:
 - ``meta/xx.yml`` - can you move them 'as is' to the new PR, or do you rewrite them somehow? 
- sanity ignore - I think these are clean removes. In the new PR, you may need to add similar (but not identical) entries into the new collection's ignore?
- tests - my guess is it's a rewrite in the new PR to account for FQCN etc?

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 5 above)
Comment (felixfontein) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: It should be removed from the old collection. This part is about removing things from the old collection, so I think it should say "remove". It should also say "removing" instead of "moving" in the beginning IMO.

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 6 above)
Comment (felixfontein) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: * `meta/runtime.yml` is tricky - it depends a lot on the content. Usually, new entries have to be added, and potentially some entries have to be removed (but probably not moved).
* sanity ignore: it's probably a move, but definitely a remove in the source collection. Once ansible-test can be configured per collection, it might not be a simple re-add in the target collection anymore, but right now it is.
* tests: it's usually copying them, and adjusting all FQCNs (same as in the plugins/modules themselves, their examples and references contain FQCNs that should potentially be updated).

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 7 above)
Comment (felixfontein) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: I'd argument for `remove` as before. It should definitely be the same word though, no matter which one :)

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 8]
```diff
@@ -470,24 +470,58 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+Adding the content to the new collection
+-----------------------------------------
+
+Create a PR in the new collection to:
+
+#. Add ALL the files removed in first PR (from the old collection).
+#. If it is an action plugin, include the corresponding module with documentation.
+#. If it is a module, check if it has a corresponding action plugin that should move with it.
+#. Check ``meta/ `` for relevant updates to ``action_groups.yml`` and ``runtime.yml`` if they exist.
+#. Carefully check ``tests/integration`` and ``tests/units``.
```
Comment (bmillemathias) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Add checks related to the files added into ``tests/integration`` and ``tests/units``. ?

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 9]
```diff
@@ -470,24 +470,70 @@ Collection versions use `Semantic Versioning <https://semver.org/>`_ for version
...
+
+
+Removing the content from the old collection
+----------------------------------------------
+
+Create a PR against the old collection repo to remove the modules, module_utils, plugins, and docs_fragments related to this migration:
+
+#. If you are moving an action plugin, include the corresponding module with documentation.
+#. If you are moving a module, move any corresponding action plugin that should stay with it.
+#. Remove any entries about removed plugins from ``meta/action_groups.yml`` and ``meta/runtime.yml``. Ensure they are added into the new repo.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. Remove any entries about removed plugins from ``meta/runtime.yml``. Ensure they are added into the new repo.
```
Action groups are now defined inside `meta/runtime.yml` as well. These entries cannot be moved that simply though, since action groups are currently limited to a certain hard-coded list of group(s) allowed per collection. (It's best if modules mentioned in action groups are not moved at all.)

**Comment by gundalow** [comment]
Once merged could you please update

1.  `When moving modules between collections` section in https://github.com/ansible-collections/overview/blob/HEAD/collection_requirements.rst

2. Delete https://github.com/ansible-collections/overview/blob/HEAD/README.rst#q-what-should-i-do-to-move-plugins-across-collections-during-migration and replace with a link to this repo
